### PR TITLE
[ENG-2262] Remove search button from navbar

### PR DESCRIPTION
**Pull request by baylee-d**
- Ticket: https://openscience.atlassian.net/browse/ENG-2262
- Feature flag: n/a

## Purpose

The navbar wraps weirdly for long names at certain screen sizes. This PR will fix that issue.

## Summary of Changes

- Remove search bar from the navbar

## Side Effects

`N/A`

## QA Notes

This should fix the navbar wrapping weird at screen sizes. This was tested for `Real World Evidence Registry`, but it will still break for longer names. The long-term solution is to eventually put everything in a hamburger menu, but this should fix the issue for now with the providers we know we'll have.

### CODE REVIEW

File: lib/registries/addon/components/registries-navbar/template.hbs [hunk 1]
```diff
@@ -64,23 +69,6 @@
...
                 </nav.links.secondary>
             {{/if}}
-        </nav.section>
-
-        {{! Right Side (Search, Buttons, Gravatar, Dropdown) }}
-        <nav.section data-test-right>
-
-            <nav.item
-                data-test-search-bar='1'
-                @classNames={{local-class 'HideOnMobile SearchBar'}}
```
Comment (fabmiz) [inline lib/registries/addon/components/registries-navbar/template.hbs:74]: Should we also delete `SearchBar` css class?

File: lib/registries/addon/components/registries-navbar/template.hbs [hunk 2]
```diff
@@ -64,23 +69,6 @@
...
-            >
-                <IconInput
-                    data-analytics-name='Search box'
-                    @ariaLabel={{t 'navbar.search_bar'}}
-                    @icon='search'
-                    @dark={{true}}
-                    @enter={{this.search}}
```
Comment (fabmiz) [inline lib/registries/addon/components/registries-navbar/template.hbs:81]: Should we delete this `action`?

File: lib/registries/addon/components/registries-navbar/template.hbs [hunk 3]
```diff
@@ -252,18 +240,3 @@
         </nav.section>
     </nav.container>
 </Navbar>
-
-{{#if (media 'isMobile')}}
-    <Container
-        data-test-search-bar-mobile='1'
-        local-class='SearchDropdown'
```
Comment (fabmiz) [inline lib/registries/addon/components/registries-navbar/template.hbs:259]: Delete this class as well??

**Review by fabmiz — COMMENTED** [review]
Some leftover css classes and actions? Other looks great!

**Review by fabmiz — APPROVED** [review]
💯

**Review by futa-ikeda — APPROVED** [review]

**Pull request merged**